### ISSUE TITLE
Why AGPL?

### ISSUE THREAD

**Issue by tedyoung** [asciimoo/wuzz#78]
Are you planning on commercializing this tool? Otherwise I'm unclear as to why AGPL was selected -- this would bar me from using it at work.

**Comment by stef** [comment]
why does the AGPL bar you from using it at work? do you have some special work contract that prohibits you to use agpl services online? if so,you might want to look for a more ethical employer.

**Comment by Benaiah** [comment]
I understand that AGPL may cause trouble when the licensed code is used as part of a larger application, as the AGPL wouldn't allow you to run a closed-source SAAS with AGPL components, for instance. I don't really understand the concern when it comes to wuzz, though, as wuzz is entirely user-facing (at least for the time being) - there's not really a way to use it as part of a larger product. As @asciimoo put it in IRC once, "headless wuzz is curl".

**Comment by tedyoung** [comment]
Because lawyers? It's a barrier and whether the barrier is valid or not, why put up the barrier in the first place? GPL would be fine and serve the same purpose, no?

Again, I'd understand the desire for AGPL if commercialization was happening, but the entire point of AGPL is stated as:

> The purpose of the GNU Affero GPL is to prevent a problem that affects developers of free programs that are often used on servers.

Since this would never be used on servers as part of a service, the AGPL seems to needlessly restrict its usage for those who work at companies who are perhaps overly paranoid about AGPL.

**Comment by stef** [comment]
in this case the AGPL == GPL it seems. the only problem seems to be with as you say your laywers, maybe the root cause needs fixing, not the symptom?

**Comment by tedyoung** [comment]
> the only problem seems to be with as you say your laywers, maybe the root cause needs fixing, not the symptom?

I totally agree, but given the size of my company, that's not really an option. Personally, I'd love for them to be less paranoid about it, but that's not the reality I live in.

If there's no desire to change it, that's cool, it's not my project, but it'd be unfortunate as I'd love to contribute.

**Comment by Benaiah** [comment]
@tedyoung sorry, I'm a little unclear - is the legal issue at your company a problem with you *using* wuzz or *contributing* to it?

**Comment by tedyoung** [comment]
@Benaiah Unfortunately at my company it's a problem to even download it (without prior legal approval). So I can neither download, use, nor contribute to code that is AGPL.

**Comment by Benaiah** [comment]
@tedyoung that's quite unfortunate; sorry you're in that predicament.

I totally understand your position - the AGPL in particular is pretty scary from a corporate lawyer's perspective. In this case it shouldn't be an issue, but lawyers are lawyers and nixing AGPL software altogether is a lot less work than actually understanding the effects of it in each particular case.

I've talked with @asciimoo about it, and we definitely don't want to relicense to a permissive (MIT/BSD style) license. He *might* be open to GPL relicensing, but that would require getting signoff from all current contributors, and we'd still have trouble with corps that ban all GPL-family software, which in my experience is more common than allowing GPL but not AGPL. I'm honestly surprised that your company makes a distinction between the GPL and AGPL - that's a more detailed understanding of free licenses than I'd expect from a corporate legal department.

Personally, I think a relicense is unlikely. The final decision is down to @asciimoo, of course, I just don't think that getting signoff from all current contributors and relicensing all the code is feasible for us.
